### PR TITLE
Add Babel for ES6 support

### PR DESCRIPTION
**Pull request by Kegsay**
- Code now lives in `/src` : you can use ES6.
- `npm run compile` to produce `/lib` output.
- Added as a `prepublish` hook so when we `npm publish` it will compile correctly before release.
- Watch and rebuild via `npm run start`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +48,10 @@
     "request": "^2.53.0"
   },
   "devDependencies": {
+    "babel": "^6.5.2",
+    "babel-cli": "^6.18.0",
+    "babel-polyfill": "^6.20.0",
```
Comment (richvdh) [inline package.json]: should this not be a dependency rather than a `devDependency`?

File: package.json [hunk 2]
```diff
@@ -7,11 +7,12 @@
     "test": "istanbul cover --report cobertura --config .istanbul.yml -i \"lib/**/*.js\" jasmine-node -- spec --verbose --junitreport --captureExceptions",
     "check": "jasmine-node spec --verbose --junitreport --captureExceptions",
     "gendoc": "jsdoc -r lib -P package.json -R README.md -d .jsdoc",
+    "compile": "babel -d lib src",
```
Comment (richvdh) [inline package.json]: can you make this write a source map, please? (`-s` ?)

Comment (richvdh) [inline package.json]: we use `build` for this in react-sdk. Any reason to break that convention?

File: package.json (hunk 1 above)
Comment (Kegsay) [inline package.json]: The polyfill in particular? Probably actually, yeah.

File: package.json (hunk 2 above)
Comment (Kegsay) [inline package.json]: Done.

File: package.json (hunk 1 above)
Comment (Kegsay) [inline package.json]: Done.

**Comment by Kegsay** [comment]
> should we include src in the npm package, so people who target ES6 in their apps don't need to use the transpiled version?

Done.

> y no `start` target? think you just need to repeat the babel command with a -w.

What are you talking about? What's a "start target"?

**Comment by richvdh** [comment]
> What are you talking about? What's a "start target"?

well, the react-sdk has a `start` script which starts a watch/rebuild doofer. Likewise vector-web.

**Comment by Kegsay** [comment]
If it's as easy as adding `-w` then fine, I'll do it. I don't want to make a huge PR to add this though if it's a gigantic PITA.

**Comment by Kegsay** [comment]
Seems to work. Done.

File: package.json (hunk 2 above)
Comment (richvdh) [inline package.json]: half done.

**Comment by Kegsay** [comment]
`build` is already in use

**Pull request merged**